atc/engine/engine.go

```diff
@@ -392,3 +395,42 @@ func (c *engineCheck) clearRunState() {
 	id := fmt.Sprintf("check:%v", c.check.ID())
 	c.trackedStates.Delete(id)
 }
+
+func (c *engineCheck) trackStarted(logger lager.Logger) {
+	found, err := c.check.Reload()
```

Comment:
I wonder how valuable it will actually be to have the check pending duration. If we have to requery the db for this info maybe it isn't worth it?